### PR TITLE
feat: redesign fitness scoring to be value-oriented

### PR DESCRIPTION
**Pull request by Danielsio**
## Summary

- Rewrites the FitnessScore algorithm to answer **"how good a value is this car?"** instead of "how close to the search ceiling?"
- Fixes the admin UI bug where `fitness_score` (0–10 scale) was treated as a 0–1 fraction and displayed as a percentage

### Scoring dimensions (`internal/scoring/scoring.go`)

| Dimension | Old behavior | New behavior |
|-----------|-------------|-------------|
| **Price** | Higher price within range = lower score | **Inverted**: cheaper = better via `sqrt(1 - ratio)` |
| **Km** | Flat ratio against `maxKm` | **Age-adjusted**: blends expected km/year (15k Israeli avg) with cap score |
| **Hand** | Fixed ladder (1→1.0, 2→0.7, 3→0.4) | Same ladder + **age bonus** (up to +0.15 for older cars) |
| **Year** | Concave curve, harsh on older years | **Floor at 0.3** + sqrt curve — less double-counting with price/km |
| **Engine** | No change | No change |

### Concrete impact

Honda Accord 2014 (39,999 NIS, 82k km, hand 2) with search PriceMax=50k, YearMin=2012–2020:

| Dimension | Before | After |
|-----------|--------|-------|
| Price | 0.28 | 0.47 |
| Km | 0.74 | 0.60 |
| Hand | 0.70 | 0.82 |
| Year | 0.38 | 0.65 |
| **Total** | **~5.3** ("weak") | **~6.4** ("good") |

### Admin UI (`web/src/pages/AdminPage.tsx`)

- Badge thresholds: `0.7/0.4` → `7.0/4.0`
- Display: removed `× 100` percentage, now shows `6.4` / `6.4 / 10`

### Demo scoring (`web/src/lib/scoringAlgorithm.ts`)

- Synced with backend: price inversion, age-adjusted km, year floor, hand age bonus, correct weights (35/25/20/15/5)

## What does NOT change

- DealScore / ScoreWithKm / MarketCache (market-relative, working correctly)
- Dimension weights (35/25/20/15/5)
- Score scale (0–10) and label thresholds
- Storage, API, frontend rendering pipeline

## Test plan

- [x] All 30 scoring unit tests pass (including new Honda Accord integration test)
- [x] `TestKmScore_AgeAdjusted` — validates old vs new car with same km
- [x] `TestHandScore_AgeBonus` — validates age bonus for hand > 1
- [x] `make test` passes (81.1% coverage)
- [x] `go vet` clean
- [x] TypeScript type-check clean
- [x] ESLint clean

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Improvements**
  * Smart Match scoring refined with age-aware mileage and ownership adjustments, smoother price/year curves, and normalized 0–10 scoring with updated weightings.
* **Bug Fixes**
  * Admin listings now show fitness score as a decimal 0–10 (e.g., "X.X / 10") with updated threshold behavior.
* **Tests**
  * Expanded and stabilized scoring tests covering age-related, edge-case, and end-to-end scenarios.

### CODE REVIEW

File: internal/scoring/scoring.go [hunk 1]
```diff
@@ -274,57 +263,93 @@ func FitnessScoreDetailed(p FitnessParams) FitnessResult {
...
+		// Cap-based: how far below the user's km limit.
+		capRatio := float64(km) / float64(maxKm)
+		capScore := clamp01(1.0 - math.Pow(clamp01(capRatio), 1.5))
+		return ageAdjustKmBlend*ageScore + (1-ageAdjustKmBlend)*capScore
 	}
-	ratio := float64(km) / float64(ref)
-	s := 1.0 - math.Pow(clamp01(ratio), curveKm)
-	return clamp01(s)
+	return ageScore
 }
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply internal/scoring/scoring.go:326]: Fixed in 7cc1a1d. Added a `carYear > 0` guard — when year is missing, `kmScore` now falls back to cap-only scoring (if `maxKm > 0`) or returns `NaN` (omit dimension) if there's no signal at all. Added `TestKmScore_ZeroCarYear` covering both paths.

File: web/src/lib/scoringAlgorithm.ts [hunk 2]
```diff
@@ -35,34 +35,55 @@ export function scoreListingAgainstSearch(
...
+  const carAge = Math.max(1, now - listing.year);
+  const expectedKm = carAge * AVG_KM_PER_YEAR;
+  const ageScore = clamp01(1 - Math.pow(clamp01(listing.mileage / expectedKm), 1.2));
+  let mileageFactor: number;
+  if (search.mileage_max > 0) {
+    const capScore = clamp01(1 - Math.pow(clamp01(listing.mileage / search.mileage_max), 1.5));
+    mileageFactor = 0.6 * ageScore + 0.4 * capScore;
+  } else {
+    mileageFactor = ageScore;
+  }
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply web/src/lib/scoringAlgorithm.ts:87]: Fixed in 7cc1a1d. The demo now short-circuits `mileageFactor = NaN` when `listing.mileage <= 0`, and the combined score loop skips NaN dimensions and redistributes weights — matching the Go backend behavior exactly.

**Comment by Danielsio** [comment]
@coderabbitai review

File: internal/scoring/scoring.go [hunk 3]
```diff
@@ -274,57 +276,100 @@ func FitnessScoreDetailed(p FitnessParams) FitnessResult {
...
 		return 0.5
 	}
 	ratio := float64(price) / float64(priceMax)
-	s := 1.0 - math.Pow(clamp01(ratio), curvePrice)
-	return clamp01(s)
+	if ratio >= 1.0 {
+		return 0.0
+	}
+	return math.Sqrt(1.0 - ratio)
 }
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply internal/scoring/scoring.go:292]: Acknowledged — the `priceMax <= 0` branch in `priceScore` is indeed dead code since the dimension wrapper short-circuits to NaN. Keeping it as a defensive guard for direct callers (like `TestPriceScore`) so the function remains testable in isolation. Low priority, fine to defer.

File: web/src/lib/scoringAlgorithm.ts [hunk 4]
```diff
@@ -35,42 +35,77 @@ export function scoreListingAgainstSearch(
...
+  } else {
+    const expectedKm = carAge * AVG_KM_PER_YEAR;
+    const ageScore = clamp01(1 - Math.pow(clamp01(listing.mileage / expectedKm), 1.2));
+    if (search.mileage_max > 0) {
+      const capScore = clamp01(1 - Math.pow(clamp01(listing.mileage / search.mileage_max), 1.5));
+      mileageFactor = 0.6 * ageScore + 0.4 * capScore;
+    } else {
+      mileageFactor = ageScore;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply web/src/lib/scoringAlgorithm.ts]: Fixed in 50d1228. Added `hasAge = listing.year > 0` guard mirroring the Go `kmScore` fix — when year is missing, km scoring falls back to cap-only or NaN.
